main.tf

```diff
@@ -602,6 +662,35 @@ resource "aws_subnet" "intra" {
   )
 }
 
+################################################################################
+# Firewall subnet
+################################################################################
+resource "aws_subnet" "firewall" {
+  count = var.create_vpc && length(var.firewall_subnets) > 0 && (false == var.one_nat_gateway_per_az || length(var.firewall_subnets) >= length(var.azs)) ? length(var.firewall_subnets) : 0
```

Comment:
I'm a little confused where the `var.one_nat_gateway_per_az` fits into this. From [AWS' post on Deployment Models for Network firewall](https://aws.amazon.com/blogs/networking-and-content-delivery/deployment-models-for-aws-network-firewall/), if you're using a NAT then they still expect that the NATs will be in the protected subnets, which will connect to the Network Firewall

Reply:
i think i'll follow what you did to make it simpler, i was overthinking and following the count from `aws_subnet.public`

I'll try doing a reverse PR from your branch to mine for this.